contracts/accountants/GenericAccountant.vy

```diff
@@ -50,59 +97,125 @@ MAX_SHARE: constant(uint256) = 7_500
 SECS_PER_YEAR: constant(uint256) = 31_556_952  # 365.2425 days
 
 
-# STORAGE #
+### STORAGE ###
+
+# Address in charge of the accountant.
 fee_manager: public(address)
+# Address to become the fee manager.
 future_fee_manager: public(address)
-fees: public(HashMap[address, Fee])
-refund_ratios: public(HashMap[address, uint256])
-asset: public(address)
+# Address to distribute the accumulated fees to.
+fee_recipient: public(address)
 
+# Mapping of vaults that this serves as an accountant for.
+vaults: public(HashMap[address, bool])
+# Default config to use unless a custom one is set.
+default_config: public(Fee)
+
+# Mapping of strategy to bool if it has custom fees.
+custom: public(HashMap[address, bool])
+# Mapping strategy => custom Fee config
+fees: public(HashMap[address, Fee])
 
 @external
-def __init__(asset: address):
-    self.fee_manager = msg.sender
-    self.asset = asset
+def __init__(
+    fee_manager: address, 
+    fee_recipient: address,
+    default_management: uint16, 
+    default_performance: uint16, 
+    default_refund: uint16, 
+    default_max: uint16
+):
+    """
+    @notice Initialilze the accountant and default fee config.
+    @param fee_manager Address to be in charge of this accountant.
+    @param default_management Default annual management fee to charge.
+    @param default_performance Default performance fee to charge.
+    @param default_refund Default refund ratio to give back on losses.
+    @param default_max Default max fee to allow as a perfent of gain.
+    """
+    assert fee_manager != empty(address), "ZERO ADDRESS"
+    assert fee_recipient != empty(address), "ZERO ADDRESS"
+    assert default_management <= self._management_fee_threshold(), "exceeds management fee threshold"
+    assert default_performance <= self._performance_fee_threshold(), "exceeds performance fee threshold"
+
+    # Set initial addresses
+    self.fee_manager = fee_manager
+    self.fee_recipient = fee_recipient
+
+    # Set the default fee config
+    self.default_config = Fee({
+        management_fee: default_management,
+        performance_fee: default_performance,
+        refund_ratio: default_refund,
+        max_fee: default_max
+    })
+
+    log UpdateDefaultFeeConfig(self.default_config)
 
 
 @external
 def report(strategy: address, gain: uint256, loss: uint256) -> (uint256, uint256):
```

Comment:
Yes i agree. The main downside to this is you cant set custom fees for a strategy based on the vault. I thought about adding a vault => strategy => custom config mapping as well, but thought the extra gas to check that each time wasnt worth it considering given how little it would likely be used.

Maybe can use a Enum type of flag to specify which config to use. Could help make the sloads more effecient for all the configs.

Reply:
strategies dont have a vault parameter anymore. Since they can be added to any number of vaults or used without them

Reply:
should be just one more sload, right. 

Its probably worth it to not automatically use the same config for a strategy if its added to multiple vaults just because a custom one should be set for one of the vaults